extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,205 @@
+  {
+    return CloseableIterators.withEmptyBaggage(readAsList().iterator());
+  }
+
+  List<InputRow> readAsList()
+  {
+    try {
+      return parseMetricsData(MetricsData.parseFrom(source.getBuffer()));
+    }
+    catch (InvalidProtocolBufferException e) {
+      throw new ParseException(e, "Protobuf message could not be parsed");
+    }
+  }
+
+  private List<InputRow> parseMetricsData(final MetricsData metricsData)
+  {
+    return metricsData.getResourceMetricsList()
+        .stream()
+        .flatMap(resourceMetrics -> {
+          Map<String, Object> resourceAttributes = resourceMetrics.getResource()
+              .getAttributesList()
+              .stream()
+              .collect(Collectors.toMap(kv -> resourceAttributePrefix + kv.getKey(), kv -> kv.getValue().getStringValue()));
+          return resourceMetrics.getInstrumentationLibraryMetricsList()
+              .stream()
+              .flatMap(libraryMetrics -> libraryMetrics.getMetricsList()
+                  .stream()
+                  .flatMap(metric -> parseMetric(metric, resourceAttributes).stream()));
+        })
+        .collect(Collectors.toList());
+  }
+
+  private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAttributes)
+  {
+    List<InputRow> inputRows;
+    String metricName = metric.getName();
+    switch (metric.getDataCase()) {
+      case INT_SUM: {
+        inputRows = new ArrayList<>(metric.getIntSum().getDataPointsCount());
+        metric.getIntSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseIntDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case INT_GAUGE: {
+        inputRows = new ArrayList<>(metric.getIntGauge().getDataPointsCount());
+        metric.getIntGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseIntDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case SUM: {
+        inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
+        metric.getSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case GAUGE: {
+        inputRows = new ArrayList<>(metric.getGauge().getDataPointsCount());
+        metric.getGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      // TODO Support and SUMMARY metrics
+      default:
+        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
+    }
+    return inputRows;
+  }
+
+  private InputRow parseIntDataPoint(IntDataPoint dataPoint, Map<String, Object> resourceAttributes, String metricName)
+  {
+    int capacity = resourceAttributes.size() + dataPoint.getLabelsCount() + 2;
```

Comment:
any reason why we don't count data point attributes here? I'm not sure we can assume that the switch from labels to attributes happens in the same version as the switch from int to numberdatapoint